### PR TITLE
resolve possible null exception

### PR DESCRIPTION
**Pull request by xichan-ms**
[AB#350535](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/350535)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6918)

### CODE REVIEW

**Review by yufeih — CHANGES_REQUESTED** [review]
Enable `<nullable>` in this project and examine all null problems reported by the compiler

File: src/Microsoft.Docs.LearnValidation/DependencyMap/DependencyItem.cs [hunk 1]
```diff
@@ -9,13 +9,13 @@ namespace Microsoft.Docs.LearnValidation
     [JsonObject(NamingStrategyType = typeof(SnakeCaseNamingStrategy))]
     public class DependencyItem
     {
-        public string FromFilePath { get; set; }
+        public string? FromFilePath { get; set; }
```
Comment (OsmondJiang) [inline src/Microsoft.Docs.LearnValidation/DependencyMap/DependencyItem.cs:12]: can we just make them non-nullable?

Comment (xichan-ms) [reply src/Microsoft.Docs.LearnValidation/DependencyMap/DependencyItem.cs:12]: ok, will modify.

File: src/Microsoft.Docs.LearnValidation/Utils/Utility.cs [hunk 2]
```diff
@@ -7,19 +7,16 @@ namespace Microsoft.Docs.LearnValidation
 {
     public static class Utility
     {
-        private static readonly Regex s_sshGitUrlRegex = new(@"git@(?<host>.+?):(?<userName>.+?)\/(?<repoName>.+)\.git", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        private static readonly Regex SshGitUrlRegex = new(@"git@(?<host>.+?):(?<userName>.+?)\/(?<repoName>.+)\.git", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
Comment (yufeih) [inline src/Microsoft.Docs.LearnValidation/Utils/Utility.cs:10]: revert this change: it should cause a build warning: https://dev.azure.com/ceapex/Engineering/_build/results?buildId=261090&view=logs&j=a29a2371-0793-5e14-39b9-20eb0beef0db&t=61e53b64-9dac-5231-7eb0-82834e091e5f&l=51

File: src/Microsoft.Docs.LearnValidation/Models/LegacyManifestOutputItem.cs [hunk 3]
```diff
@@ -10,13 +10,13 @@ public class LegacyManifestOutputItem
     {
         // output path relative to site base path
         [JsonProperty("relative_path")]
-        public string RelativePath { get; set; }
+        public string RelativePath { get; set; } = string.Empty;
```
[the remark this answers is not in the document]
Comment (xichan-ms) [reply src/Microsoft.Docs.LearnValidation/Models/LegacyManifestOutputItem.cs:13]: OK

File: src/Microsoft.Docs.LearnValidation/Microsoft.Docs.LearnValidation.csproj [hunk 4]
```diff
@@ -4,7 +4,11 @@
     <OutputType>Library</OutputType>
     <TargetFramework>net5.0</TargetFramework>
     <EnableNETAnalyzers>false</EnableNETAnalyzers>
-    <nullable>disable</nullable>
+    <nullable>enable</nullable>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
[the remark this answers is not in the document]
Comment (xichan-ms) [reply src/Microsoft.Docs.LearnValidation/Microsoft.Docs.LearnValidation.csproj:10]: ok

File: src/Microsoft.Docs.LearnValidation/Utils/Utility.cs (hunk 2 above)
Comment (xichan-ms) [reply src/Microsoft.Docs.LearnValidation/Utils/Utility.cs:10]: ok

File: src/Microsoft.Docs.LearnValidation/DependencyMap/DependencyItem.cs [hunk 5]
```diff
@@ -9,13 +9,13 @@ namespace Microsoft.Docs.LearnValidation
...
+        public string FromFilePath { get; set; } = string.Empty;
 
-        public string ToFilePath { get; set; }
+        public string ToFilePath { get; set; } = string.Empty;
 
-        public string DependencyType { get; set; }
+        public string DependencyType { get; set; } = string.Empty;
 
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
-        public string Version { get; set; }
```
Comment (herohua) [inline src/Microsoft.Docs.LearnValidation/DependencyMap/DependencyItem.cs:19]: This can be `null` (see `JsonProperty`)?

File: src/Microsoft.Docs.LearnValidation/LearnValidationLogger.cs [hunk 6]
```diff
@@ -8,13 +8,7 @@ namespace Microsoft.Docs.LearnValidation
 {
     public class LearnValidationLogger
     {
-        /// <summary>
-        /// Delegate to write to .errors.log file
-        /// </summary>
-        private readonly Action<LearnLogItem> _writeLog;
-        private readonly HashSet<string> _filesWithError;
-
-        private static readonly Dictionary<LearnErrorCode, string> s_errorMessageMapping = new()
```
Comment (herohua) [inline src/Microsoft.Docs.LearnValidation/LearnValidationLogger.cs:17]: these lines seem no need to change

Comment (xichan-ms) [reply src/Microsoft.Docs.LearnValidation/LearnValidationLogger.cs]: Fixed, due to the false positive of visual studio.

File: src/Microsoft.Docs.LearnValidation/DependencyMap/DependencyItem.cs (hunk 5 above)
Comment (xichan-ms) [reply src/Microsoft.Docs.LearnValidation/DependencyMap/DependencyItem.cs:19]: OK!

**Review by yufeih — APPROVED** [review]
